docs/upcoming_changes/9309.change.rst

```diff
@@ -0,0 +1,8 @@
+Make test listing not invoke CPU compilation.
+---------------------------------------------
```

Comment:
@kc611 @esc Should these be `"` or does it make no difference?

Reply:
I think they should be either `"` or `=` or something else that is unused. We maybe need to revisit this. RST uses the convention, that you can use any character for underlining, but it must be consistent.

Looking at the last two releases:

https://github.com/numba/numba/tree/main/docs/source/release

https://raw.githubusercontent.com/numba/numba/main/docs/source/release/0.58.1-notes.rst

and

https://raw.githubusercontent.com/numba/numba/main/docs/source/release/0.58.0-notes.rst

0.58.0 used the following structure:

First level: `---` (hyphen)
Second level: `~~~` (tilde)
Third level: `""""` (quotes)

But 0.58.1 uses:

First level: `---` (hyphen)
Second level: `~~~` (tilde)
Third level: `====` (equals)

If we look at the upcoming changes in:

https://github.com/numba/numba/tree/main/docs/upcoming_changes

It's a mix of `=` and `"` unfortunately.

The ideal solution IMHO would be:

```
=========
Version xxx
=========

Highlights
=======

Feature X was implemented
----------------------------

Description of feature X

Bug Fixes
=======

Bug Y was fixed
----------------

Description of bug Y fixed

Pull-Requests
==========

* 1
* 2
* 3

Authors
======

* a
* b
* c

```

So, first level headings use double equals `=` (top and bottom, you can also uses overline and underline, but must be same length and same character). Second level heading uses underline `=` equals and third level heading to use `-` hyphens. Thus, the snippet titles would use `-` hyphen which, perhaps, is most natural for contributors. Only the docs about snippets, the PR-list generation script and the (soon to come) automatic checking action would need to be changed. Existing release notes can stay as is as each release note is in its own file.

Reply:
Thanks for this explanation @esc, perhaps this should be captured in #9311 with view of enforcing it as part of the changes suggested in that task?

Reply:
https://github.com/numba/numba/issues/9311#issuecomment-1834341155